From 3a792990dcdde27de1cd94292ce55a367261dce2 Mon Sep 17 00:00:00 2001
From: Henry Beberman <henry.beberman@microsoft.com>
Date: Thu, 9 Apr 2020 14:52:16 -0700
Subject: [PATCH] Disable argument to mount_cgroup_controllers

Cgroup controller parsing was leading to segfaults while initalizing
systemd on certain configurations.

Upstream systemd has deprecated the argument to this function
in pull request #10785 "core: remove JoinControllers= configuration
setting"

Squelching the argument here and using the defaults is the simplest
resolution until we move forward systemd versions.
---
 src/core/mount-setup.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/core/mount-setup.c b/src/core/mount-setup.c
index 16880e6157..085204d01e 100644
--- a/src/core/mount-setup.c
+++ b/src/core/mount-setup.c
@@ -239,7 +239,8 @@ int mount_cgroup_controllers(char ***join_controllers) {
 
         /* Mount all available cgroup controllers that are built into the kernel. */
 
-        if (!has_argument)
+        // Use the defaults. The argument to this mount_cgroup_controllers has been
+        // deprecated upstream in systemd pull request #10785 so override it here always.
                 /* The defaults:
                  * mount "cpu" + "cpuacct" together, and "net_cls" + "net_prio".
                  *
-- 
2.17.1

